universal-application-tool-0.0.1/app/views/components/TextFormatter.java

```diff
@@ -43,9 +45,22 @@
   private static final String BULLETED_ITEM = "* ";
 
   public static ImmutableList<DomContent> createLinksAndEscapeText(String content) {
-    List<Url> urls = new UrlDetector(content, UrlDetectorOptions.Default).detect();
+    // JAVASCRIPT option avoids including surrounding quotes or brackets in the URL.
+    List<Url> urls = new UrlDetector(content, UrlDetectorOptions.JAVASCRIPT).detect();
+
     ImmutableList.Builder<DomContent> contentBuilder = ImmutableList.builder();
     for (Url url : urls) {
+      try {
+        // While technically they could be part of the URL, trailing punctuation
+        // is more likely to be part of the surrounding text, so we strip.
+        url = Url.create(StringUtils.stripEnd(url.getOriginalUrl(), ".?!,:;"));
```

Comment:
Should we include any of the characters below here?

` < >  [ ] { } 

You may have avoided some of these because they can be in URLs and/or are unlikely to be used as trailing punctuation. Looks like UrlDetectorOptions.JAVASCRIPT detects some of these, so just wanna make sure we're not missing anything else.

Reply:
I avoided characters that should be already be covered by the JAVASCRIPT option <>[]{} plus single and double quotes. Overall seemed safest to just go after characters that are likely to be part of the surrounding text. The single opening quote you mentioned is an interesting case -- seems unlikely to be part of an URL, or to follow an URL in surrounding text. I don't have a strong feeling, but my inclination would be to err on the side of keeping this fixup hack more constrained.

Reply:
That SGTM! Thanks for the explanation.